### ISSUE TITLE
unclean reactor error from test_newStyleCache on with glib2 reactor

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2866]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#2866|
|Type|defect|
|Created|2007-11-05 02:33:28Z|

===============================================================================
[ERROR]: twisted.test.test_pb.NewStyleCachedTestCase.test_newStyleCache

Traceback (most recent call last):
Failure: twisted.trial.util.DirtyReactorAggregateError: Reactor was unclean.
Selectables:
<Broker #0 on 0>
===============================================================================

<details><summary>Searchable metadata</summary>

```
trac-id__2866 2866
type__defect defect
reporter__exarkun exarkun
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1194230008000000 1194230008000000
changetime__1194270949000000 1194270949000000
version__None None
owner__ 
cc__therve
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_1"></a>|@therve commented|
|-|-|

(In [21576]) Remove the unclean error.

Refs [#2866](https://github.com/twisted/twisted/issues/2866)

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve commented|
|-|-|

Ready to review in test-new-style-cached-unclean-2866.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)|@therve removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set owner to @therve|
|-|-|

```self.server.factory.protocolInstance.transport.loseConnection``` - ouch.  That really makes me want to think about simplifying something - but I'm not sure what.

typo in ConnectionNotifyServerFactory class docstring - _Deffered_.  Other than that, looks good.  Tested w/ glib2 on my local machine, got the error without the branch, didn't with it.  So please merge once you fix that docstring.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_5"></a>|@therve commented|
|-|-|

(In [21581]) Typos.

Refs [#2866](https://github.com/twisted/twisted/issues/2866)

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_6"></a>|@therve set status to `closed`|
|-|-|

(In [21582]) Merge test-new-style-cached-unclean-2866

Author: therve
Reviewer: exarkun
Fixes [#2866](https://github.com/twisted/twisted/issues/2866)

Fix an unclean error in test_newStyleCache with glib2 reactor.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
